main_exercise/server/routes/api.js

```diff
@@ -0,0 +1,36 @@
+//Define your endpoints here (this is your "controller file")
+//
+// import express from "express";
+const express = require("express");
+// import ItemManager from "../services/item_manager.js";
+const ItemManager = require("../services/item_manager.js");
+
+const router = express.Router();
+
+router.use(async (req, res, next) => {
+    console.log("A request was made to the API");
+    console.log(req.body)
+    next();
+});
+const itemManager = new ItemManager();
+router.get('/todos', async (req, res) => {
+    res.send(await itemManager.getTodoList(req,res));
+});
+router.post('/todo', async (req, res) => {
+    await itemManager.addTodo(req.body.todo);
+    res.status(200).json(req.body.todo);
+});
+router.put('/todo', async (req, res) => {
```

Comment:
```suggestion
router.put('/todos/:id', async (req, res) => {
const { id } = req.params;
const { status } = req.body;
await itemManager.updateStatus(id, status);
```

You now have an ID for each task, therefore it's better to change the url so you'll send the task ID and then fetch the task to update based on it's id instead of it's name
https://www.codecademy.com/article/what-is-crud

Reply:
Updated for ex6